webpack.config.js

```diff
@@ -2,10 +2,47 @@
  * External dependencies
  */
 
+const path = require( 'path' );
 const glob = require( 'glob' );
 const webpack = require( 'webpack' );
 const ExtractTextPlugin = require( 'extract-text-webpack-plugin' );
 
+/**
+ * Webpack plugin that adds the containing directory of each entry point as an
+ * effective `resolve.modules` root for all module resolution within that path.
+ */
+class EntryResolvePlugin {
+	apply( compiler ) {
+		// Map entries into array of absolute paths
+		const entryRoots = Object.keys( compiler.options.entry ).map( ( entry ) => {
+			return path.dirname( path.resolve( compiler.options.entry[ entry ] ) );
+		} );
+
+		compiler.plugin( 'after-resolvers', () => {
+			compiler.resolvers.normal.apply( {
+				apply( resolver ) {
+					resolver.plugin( 'module', ( request, callback ) => {
+						// Find entry root which contains the requesting path
+						const resolvePath = entryRoots.find( ( entryRoot ) => {
+							return request.path.startsWith( entryRoot );
```

Comment:
If I'm understanding this correctly, it would break if you had two directories named like `prefix` and `prefix-and-more-stuff` - there should be a test for a path separator somewhere.